File: Projects/TransportathonHackathon.API/TransportathonHackathon.Domain/Entities/Employee.cs
using Core.Persistence.Entities;
using TransportathonHackathon.Domain.Entities.Identity;

namespace TransportathonHackathon.Domain.Entities
{
    public class Employee : Entity
    {
        public Guid AppUserId { get; set; }
        public Guid CompanyId { get; set; }
        public string FirstName { get; set; }
        public string LastName { get; set; }
        public int Age { get; set; }
        public bool IsOnTransitNow { get; set; }

        public virtual Carrier? Carrier { get; set; }
        public virtual Driver? Driver { get; set; }
        public virtual Company Company { get; set; }
        public virtual AppUser AppUser { get; set; }
    }
}

File: Projects/TransportathonHackathon.Clients/angular-ui/src/app/models/entities/employee.ts
import { Carrier } from "./carrier";
import { Driver } from "./driver";
import { Company } from "./company";
import { AppUser } from "./appUser";
import { BaseEntity } from "src/app/core/models/entities/entity";

export interface Employee extends BaseEntity{
    appUserId: string;
    companyId: string;
    firstName: string;
    lastName: string;
    age: number;
    isOnTransitNow: boolean;
    carrier: Carrier | null;
    driver: Driver | null;
    company: Company;
    appUser: AppUser;
}